apsuite/commisslib/emit_exchange/beam_sizes.py

```diff
@@ -0,0 +1,351 @@
+
+    def create_devices(self):
+        """."""
+        self.devices['evg'] = EVG()
+        self.devices['bo_ramp'] = BoosterRamp(self.params.ramp_config)
+        self.devices['bo_ramp'].load()
+        self.devices['curr_info'] = CurrInfoBO()
+        self.devices['conn_ps'] = ConnPS()
+        if not self.devices['conn_ps'].connected:
+            raise AssertionError(
+                "Can't stablish connection with power sources")
+
+        scrns = []
+        for i in range(1, 7):
+            if i <= 3:
+                scrns.append(f'TS-0{i}:DI-Scrn')
+            else:
+                scrns.append(f'TS-04:DI-Scrn-{i-3}')
+        scrn_idx = self.params.ts_screen_number-1
+        self.devices['ts_screen'] = Screen(scrns[scrn_idx])
+
+    def reset_ramp_config(self):
+        """."""
+        old_boramp = BoosterRamp(self.params.ramp_config)
+        old_boramp.load()
+        self._send_ramp_wfm(boramp=old_boramp)
+
+    def scan_emittance_exchange(self):
+        """Method to measure the beam sizes while moving the emittance exchange
+        ramp 'peak'. The results are useful to analyze the emittance exchange
+        process."""
+        boramp = self.devices['bo_ramp']
+        evg = self.devices['evg']
+        scrn = self.devices['ts_screen']
+        currinfo = self.devices['curr_info']
+        pvs = self.pvs
+        prms = self.params
+        data = self._create_dict_with_lists()
+
+        delays = _np.linspace(
+            prms.init_delay, prms.final_delay, prms.nr_points)
+
+        config_times = _np.sort(boramp.ps_normalized_configs_times)
+        emit_exc_points = config_times[-4:-1]
+        init_old, mid_old, end_old = emit_exc_points
+
+        for delay in delays:
+            # Moves the emit exchange points
+            initn, midn, endn = emit_exc_points + delay
+            boramp.ps_normalized_configs_change_time(init_old, initn)
+            boramp.ps_normalized_configs_change_time(mid_old, midn)
+            boramp.ps_normalized_configs_change_time(end_old, endn)
+            init_old, mid_old, end_old = initn, midn, endn
+
+            # Send the ramp config to the machine
+            self._send_ramp_wfm()
+            _time.sleep(5)
+
+            for i in range(prms.measures_per_point):
+                print(f"Delay = {delay}, measure {i}.")
+                evg.cmd_turn_on_injection()
+                data['measure'].append(i)
+                data['delays'].append(delay)
+                data['timestamp'].append(_time.time())
+                data['qf_wfm'].append(pvs['bo-qf-wfm'].get())
+                data['qd_wfm'].append(pvs['bo-qd-wfm'].get())
+                _time.sleep(1)
+                image = scrn.image
+                data['images'].append(image)
+                data['curr3gev'].append(currinfo.current3gev)
+
+        data['scl_factx'] = _np.abs(scrn.scale_factor_x)
+        data['scl_facty'] = _np.abs(scrn.scale_factor_x)
```

Comment:
wouldn't it be `data['scl_facty'] = _np.abs(scrn.scale_factor_y)`?

Reply:
Fixed, thanks!